### ISSUE TITLE
Failed tests with 6.9.8-7 and 7.0.5-8 (openjpeg2)

### ISSUE THREAD

**Issue by remicollet** [ImageMagick/ImageMagick#501]
```
FAIL: tests/validate-formats-disk.tap 1
FAIL: tests/validate-formats-memory.tap 1
FAIL: tests/validate-formats-map.tap 1
```
./tests/validate-formats-memory.log
```
1..1
Version: ImageMagick 6.9.8-7 Q16 x86_64 2017-05-28 http://www.imagemagick.org
Copyright: © 1999-2017 ImageMagick Studio LLC

ImageMagick Validation Suite (FormatsMemory)

[pixel-cache: memory] validate image formats in memory:
  ...
  test 221: JBG/Undefined/TrueColor/8-bits... pass
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
  test 222: JBG/Undefined/TrueColorAlpha/8-bits... pass
  ...
  test 259: JNG/JPEG/TrueColor/16-bits... pass
  test 260: JP2/Undefined/TrueColor/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 261: JP2/Undefined/TrueColorAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 262: JP2/Undefined/Grayscale/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 263: JP2/Undefined/GrayscaleAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 264: JP2/Undefined/Palette/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 265: JP2/Undefined/PaletteAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 266: JP2/Undefined/PaletteBilevelAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 267: JP2/Undefined/Bilevel/1-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 268: JP2/Undefined/ColorSeparation/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 269: JP2/Undefined/ColorSeparationAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 270: JP2/Undefined/TrueColor/10-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 271: JP2/Undefined/TrueColor/12-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 272: JP2/Undefined/TrueColor/16-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 273: J2K/Undefined/TrueColor/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 274: J2K/Undefined/TrueColorAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 275: J2K/Undefined/Grayscale/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 276: J2K/Undefined/GrayscaleAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 277: J2K/Undefined/Palette/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 278: J2K/Undefined/PaletteAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
lt-validate: irregular channel geometry not supported `/tmp/magick-31699XDQmLaCIcCy3' @ error/jp2.c/ReadJP2Image/400.
  test 279: J2K/Undefined/PaletteBilevelAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 280: J2K/Undefined/Bilevel/1-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 281: J2K/Undefined/ColorSeparation/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 282: J2K/Undefined/ColorSeparationAlpha/8-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 283: J2K/Undefined/TrueColor/10-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 284: J2K/Undefined/TrueColor/12-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 285: J2K/Undefined/TrueColor/16-bits... fail @ tests/validate.c/ValidateImageFormatsInMemory/1443.
  test 286: JPEG/Undefined/TrueColor/8-bits... pass
  ...
  test 1117: PS/Undefined/TrueColor/16-bits... pass.
  summary: 1118 subtests; 1092 passed; 26 failed.
validation suite: 2236 tests; 2210 passed; 26 failed.
not ok
FAIL: tests/validate-formats-memory.tap 1
```

**Comment by remicollet** [comment]
Notice, as related to JP2, build with `--with-openjp2` fails, while build with `--without-openjp2` passes.

**Comment by remicollet** [comment]
Could be related to c90d59f341bf8817fa9f2b372ed417eb7d8567a5 or 728697341f8c02f32c878d05902549a82e3e1af6 

BTW, golden award for useful commit message ;)

**Comment by mikayla-grace** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ http://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by remicollet** [comment]
I confirm fix is ok (tested on 7.0.5-8 + f13c6b54a879aaa771ec64b5a066b939e8f8e7f0)

Thanks

**Comment by fgeek** [comment]
Please use CVE-2017-13145 for this issue.
